PIPS-NLP/Core/StochLinearAlgebra/StochVector.h

```diff
@@ -11,10 +11,53 @@
 #include "StochVectorHandle.h"
 #include "OoqpVector.h"
 #include "SimpleVector.h"
+#include <unordered_map>
 
 #include <vector>
 
 class StochTree;
+class StochDummyVector;
+
+template <class pTree, class sTree> class VectorCompressedDummy {
+
+public:
+
+ size_t size() const {
+   return size_;
+ }
+ void clear() {
+   size_ = 0;
+   children.clear();
+ }
+
+ pTree& operator[](std::size_t idx) {
+   if (children.find(idx) == children.end()) {
+     return sTree::dummy;
+   } else {
+     return children[idx];
+   }
+ }
+
+ const pTree& operator[](std::size_t idx) const {
+   if (children.find(idx) == children.end()) {
+     return sTree::dummy;
+   } else {
+     return children[idx];
+   }
+ }
+ void push_back(const pTree child) {
+   if (child != sTree::dummy) {
+     children[size_] = child;
+   }
+   size_++;
+ }
+
+private:
+  std::unordered_map<int, pTree> children;
+  size_t size_ = 0;
+  size_t tablesize_ = 0;
```

Comment:
@michel2323  is "tablesize_" used anywhere?

Reply:
fixed!